docs/README.md

```diff
@@ -753,16 +754,138 @@ public static async handle(event: StockMoved, register: Register): Promise<void>
+the `@Reduces` decorator that also specifies the event the reducer is subscribed to. In Booster an entity looks like this:
+
+```typescript
+@Entity
+export class EntityName {
+  public constructor(readonly fieldA: SomeType, readonly fieldB: SomeOtherType /* as many fields as needed */) {}
+
+  @Reduces(SomeEvent)
+  public static reduceSomeEvent(event: SomeEvent, previousState?: EntityName): EntityName {
+    /* Return a new entity based on the previous one */
+  }
+}
+```
+
+Each time an event is generated, all reducers that are listening to it are triggered. Note that event ordering is
+preserved per entity instance. This means that if two events arrive at the same time at a reducer, the first one to
+be picked will be the one that was generated first.
+
 #### Entities naming convention
 
+Entities model your data, so name them as closely to your domain as possible. Typical entity names are nouns that
+might appear when you think about your app. In an e-commerce application, some entities would be:
+
+- Cart
+- Product
+- UserProfile
+- Order
+- Address
+- PaymentMethod
+- Stock
+
+Entities live within the entities directory of the project source: `project-root/src/entities`.
+
+```text
+project-root
+├── src
+│   ├── commands
+│   ├── common
+│   ├── config
+│   ├── entities <------ They must be here
+│   ├── events
+│   ├── index.ts
+│   └── read-models
+```
+
 #### Creating entities
 
+The preferred way to create an entity is by using the generator, e.g.
+
+```text
+boost new:entity Product --fields displayName:string description:string price:Money
+```
+
+The generator will automatically create a file called `Product.ts` with a TypeScript class of the same name under the entities directory. You can still create the entity manually. Since the generator is not doing any magic, all you need is a class decorated as `@Command`. Anyway, we recommend you always to use the generator, because it handles the boilerplate code for you.
+
+Note:
+
+> Running the entity generator with an EntityName that already exists, it will override the content of the current one. In future releases, we will display a warning before overwriting anything. Meantime, if you missed a field, just add it to the class because in Booster, all the infrastructure and data structures are inferred from your code.
+
 #### The reducer function
 
+Booster generates the reducer function as a static method of the entity class. That function is called by the framework every time that an event of the
+specified type is emitted. **This function must be pure**, which means that you shouldn't be calling third party services, reading or writing to a database, or changing something somewhere.
+
+The reducer has two parameters by default:
+
+- `event` - The event object that has triggered this reducer
+- `currentEntity?` - A possible entity generated from a previous call to the reducer. **This parameter is optional**, and will be `undefined` when an event occurred for the very first time.
```

Comment:
```suggestion
- `currentEntity?` - The current state of the entity instance that the event belongs to if it exists. **This parameter is optional** and will be `undefined` if the entity don't exist yet (For example, when you process a `NoteCreated` event that will generate the first version of a `Note` entity).
```

Reply:
Will apply this, but changing the domain to the ecommerce one, to keep with the tone

Reply:
Done https://github.com/boostercloud/booster/pull/225/commits/b2e1928eb2a349878647c4edde045f6f1d3c3552